tests/timeline.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta, date
+
+from freezegun import freeze_time
+import pytest
+
+from ics import Calendar, Event
+from ics.timezone import UTC
+
+
+@pytest.fixture
+def calendar() -> Calendar:
+    """Fixture calendar with all day events to use in tests."""
```

Comment:
I guess these are not really all-day events as they simply start at noon and end in the same instant. Would it be easier to mix different event types in the same Calendar (i.e. all-day, multi-day all-day, no specified end, fixed end or duration, maybe even with/without timezones?) and then thoroughly test that calendar?

Note: We could also infer that an Event should be all-day if its created from `date`s instead of `datetime`s, but currently the `ensure_datetime` [here](https://github.com/ics-py/ics-py/blob/e62d660/src/ics/event.py#L273) silently sets the time to midnight if its missing and ignores this distinction. Something like the following might fix that but needs further testing.
```python
if (begin is not None or end is not None or duration is not None) and "timespan" in kwargs:
    raise ValueError("can't specify explicit timespan together with any of begin, end or duration")
timespan = kwargs.pop("timespan", EventTimespan(ensure_datetime(begin), ensure_datetime(end), ensure_timedelta(duration)))
if isinstance(begin, date) and not timespan.get_effective_duration().seconds:  # TODO we should also drop microseconds to be able to ignore them here
    timespan = timespan.make_all_day()
super(Event, self).__init__(timespan, summary, *args, **kwargs)
```

Reply:
Whoops, I had intended for these to be all day events with `date` as the input, but misread them when setting them up.

Let me update them to be what they were intended to be, then see what you think.

Reply:
Note that the new functionality regarding all-day events I proposed is not yet implemented, so this still generates normal events ranging from midnight to midnight. :sweat_smile:
Currently, you'd need to explicitly call `make_all_day` on an event, but I'll probably include the code above to prevent any confusion with this.

Reply:
I've added `make_all_day` calls for now, and agree the above snippet to make this transparent makes a lot of sense.